scripts/pylib/twister/twisterlib/testinstance.py

```diff
@@ -102,8 +102,8 @@ def status(self) -> TwisterStatus:
     def status(self, value : TwisterStatus) -> None:
         # Check for illegal assignments by value
         try:
-            key = value.name if isinstance(value, Enum) else value
-            self._status = TwisterStatus[key]
+            key = value.value if isinstance(value, Enum) else value
```

Comment:
can this common code be just at one place - as a setter of `TwisterStatus` class - while there is no instance/harness/suite specifics in assignment ?

Reply:
Setters are for a property of a class, not the class itself - thus, setters are in classes containing `TwisterStatus` `Enum`, not in `Enum` itself.

You could make a kind-of-a-hack via overloading the `_missing_` method of the `Enum` to do just `self._status = TwisterStatus(value)` here, where `value` must not collide with another `Enum` value and the it will go into `_missing_` method in the `Enum` where we could do casting, checking, etc. It's not very Pythonic, I'd say.

Reply:
> Setters are for a property of a class, not the class itself 

My point is about which class property setter is implemented here, what is its actual context.
There is no difference on how `status` property is set in the current implementation of instance, harness, etc. whereas this function is mostly for `TwisterStatus.value` property correct assignment.

```
key = value.value if isinstance(value, Enum) else value
```
and all these names of value to key assignment looks a bit confusing, isn't it ?